### ISSUE TITLE
rect() : inputting negative values for width and height doesn't yield a flipped rectangle anymore...absolute values of width and height are considered

### ISSUE THREAD

**Issue by Nishchal-Bhat** [processing/p5.js#7353]
### Most appropriate sub-area of p5.js?

- [ ] Accessibility
- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [ ] DOM
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [ ] Build process
- [ ] Unit testing
- [ ] Internationalization
- [ ] Friendly errors
- [X] Other (specify if possible)

### p5.js version

v1.11.1

### Web browser and version

129.0.6668.112

### Operating system

ChromeOS

### Steps:
1. use the rect() function with negative values passed in for width and height

### Snippet:

```js

rect(100,100,40,40)
rect(100,100,-40,-40) // gives the same rectangle
```

**Comment by paul-uz** [comment]
Jsut spotted this. Pretty poor regression.

**Comment by paul-uz** [comment]
Looks like it got removed as part of this PR https://github.com/processing/p5.js/pull/7290

It got passed everyone, but clearly has a breaking change.

**Comment by davepagurek** [comment]
This was changed in https://github.com/processing/p5.js/pull/7290.

Before considering this a regression we want to fix (a negative width is arguably undefined behaviour?), @martinleopold and @ff6347 do you have opinions on what the expected behaviour here should be?

**Comment by paul-uz** [comment]
@davepagurek upon double checking, I'm not sure that is where the change happened, as the changes in that PR is for arcs. however, comparing 1.11.0 (which works) and 1.11.1, i can't see anywhere else that might of caused this regression/change

**Comment by davepagurek** [comment]
I believe `modeAdjust`, [which was modified](https://github.com/processing/p5.js/pull/7290/files#diff-bbb8a778add7280bbc1c242ccec0ad67dd853333bc0d73e947cf3f9103abeda1L7), is also is used by `rect`, and they added unit tests for rectangles too.

**Comment by davepagurek** [comment]
Also ccing the rendering stewards in case anyone has thoughts: @limzykenneth, @ChihYungChang, @teragramgius, @tuminzee, @Zarkv, @robin-haxx, @Gaurav-1306

**Comment by martinleopold** [comment]
> This was changed in #7290.
> 
> Before considering this a regression we want to fix (a negative width is arguably undefined behaviour?), @martinleopold and @ff6347 do you have opinions on what the expected behaviour here should be?

I think this was caused by my PR. It's pretty much undefined behaviour as the reference doesn't talk about negative widths and heights. Some comments in the code were mentioning "p5 supports negative width and heights for rect" but not more. 

Note that prior to my PR there were virtually no tests for the basic shape rendering/and shape modes, so no 'defined' behaviour in that regard.

Using negative width/height to flip rectangle/ellipse/arc does make sense to me though – for rectMode(CORNER) at least. I think we should revert to this behavior and use the opportunity to update the docs accordingly as well.

**Comment by limzykenneth** [comment]
Yes, it is a regression in this case. Feel free to file a PR for a fix.

**Comment by ffd8** [comment]
I just ran into this too – I usually render FFT audio bars with something like `rect(x, height, w, -freq)`, and just wondered why it stopped working?! Indeed just like `circle()` and `ellipse()` the negative value has always been an inversely drawn form since many versions ago.

**Comment by limzykenneth** [comment]
@Nishchal-Bhat @paul-uz This should be fixed in #7363 now, you can try it [here](https://editor.p5js.org/limzykenneth/sketches/oRQJl5QDK). If no other problem with this is found I'll probably do a release some time next week.

**Comment by Nishchal-Bhat** [comment]
@martinleopold @limzykenneth @paul-uz ok nice...thanks a lot for your time and effort 🙏
